src/app/_modules/product/actions.ts

```diff
@@ -0,0 +1,132 @@
+'use server';
+import type { CategoryType } from '@prisma/client';
+import type { OrderBy } from '@/app/_modules/product/types';
+import prisma from '@/lib/prisma';
+
+export const getProductsWithFilter = async ({ categoryId = '', keyword = '', sort = 'latest', page = '1' }) => {
+  try {
+    const orderBySort: {
+      [key: string]: { [key: string]: OrderBy };
+    } = {
+      latest: { createdAt: 'desc' },
+      name: { name: 'asc' },
+      priceAsc: { discountedPrice: 'asc' },
+      priceDesc: { discountedPrice: 'desc' },
+      discount: { discountRate: 'desc' },
+    };
+
+    const LIMIT = 12;
+    let where = {};
+
+    if (categoryId) {
+      where = { ...where, categoryId };
+    }
+
+    if (keyword) {
+      where = { ...where, name: { contains: keyword, mode: 'insensitive' } };
+    }
+
+    const [products, totalProducts] = await Promise.all([
+      prisma.product.findMany({
+        where,
+        orderBy: orderBySort[sort],
+        take: LIMIT,
+        skip: LIMIT * (Number(page) - 1),
+        include: {
+          brand: true,
+        },
+      }),
+      prisma.product.count({
+        where,
+      }),
+    ]);
+
+    return {
+      total: totalProducts,
+      list: products,
+    };
+  } catch (error) {
+    console.error('Fail to getProductsWithFilter', error);
+  }
+};
+
+export const getProductCategories = async () => {
+  try {
+    const categories = await prisma.category.findMany();
+    return categories;
+  } catch (error) {
+    console.error('Fail to getProductCategories', error);
+  }
+};
+
+export const getCategoryById = async (id: CategoryType) => {
+  try {
+    const category = await prisma.category.findUnique({
+      where: {
+        id,
+      },
+    });
+    return category;
+  } catch (error) {
+    console.error('Fail to getCategoryById', error);
+  }
+};
+
+export const getProduct = async (id: number) => {
+  try {
+    const data = await prisma.product.findUnique({
+      where: {
+        id,
+      },
+      include: {
+        brand: true,
+        productOptions: {
```

Comment:
Object literal may only specify known properties, and 'productOptions' does not exist in type 'ProductInclude<DefaultArgs>'.